### DISCUSSION TITLE
What other DB connectors would people like to see?

### DISCUSSION THREAD

**Discussion by andrew-strahan** [neilotoole/sq#84] [General]
Super excited to see how the community can contribute!  Happy to see comments here for what DB connectors people would most like to see! :)

**Comment by tatric** [comment]
Most of our estate uses MSSQL, it would be nice to use this tool to interact with them.

**Reply by neilotoole** [reply]
MSSQL is already supported. The driver is named “sqlserver”.

**Comment by karfau** [comment]
What about https://parquet.apache.org/ ?

**Reply by drluckyspin** [reply]
Maybe even Iceberg? https://iceberg.apache.org/

**Reply by karfau** [reply]
I guess both are interesting independently.

Iceberg is more like a server connection with the whole infra behind it, right?
With parquet you can read from or write to a local file on your disk like with csv or sqlite so you don't need to handle any auth once you have the source file.

**Comment by fabianhogger** [comment]
I think parquet would be huge for the cloud , in data analytics and other sectors it's standard for  object storage along with iceberg both would be great for migration to the cloud etc

**Comment by neilotoole** [comment]
Thanks @karfau and @fabianhogger for raising Parquet... the asks finally have tracking issues:

- **#718** — [Add Parquet as a first-class file source (DuckDB-backed)](https://github.com/neilotoole/sq/issues/718). The goal: `sq add foo.parquet`, then `sq inspect` / `sq diff` / SLQ queries work the same as they do for CSV/JSON/XLSX today. Local files, HTTPS URLs, and S3 paths in scope.
- **#719** — [Add Iceberg / Delta Lake table source (DuckDB-backed)](https://github.com/neilotoole/sq/issues/719). Covers the natural follow-on for cloud lakehouse workflows, phased: single table by path first, catalog integration (Iceberg REST, Glue, Unity Catalog) deferred to a follow-up.

Both build on the existing DuckDB driver — Parquet support is bundled out-of-the-box today (`drivers/duckdb/pragma.go`), so the work is the sq-side abstractions, not the file format itself.

### What works today as a workaround

While #718 is in flight, you can already (hopefully!) query Parquet via the DuckDB driver using raw SQL passthrough:

```bash
sq add duckdb:///tmp/scratch.duckdb
sq sql -s "SELECT * FROM read_parquet('events.parquet')"
sq sql -s "SELECT * FROM read_parquet('s3://bucket/key.parquet')"
```

No `INSTALL` / `LOAD` needed — the `parquet` extension is statically linked. See the [DuckDB driver docs](https://sq.io/docs/drivers/duckdb#bundled-extensions) for the full set of bundled extensions. It's not the same ergonomics as a first-class source (no `sq inspect`, no SLQ), but it'll unblock simple read use cases.

**Comment by trantor** [comment]
HSQLDB might be interesting. @neilotoole
